tests/script-based-pre/build-cache-fresh/cache_works.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+# Checks situations where running kani multiple times will not trigger a recompilation
+# The cases we cover here are:
+# - Exact same input being invoked 2x.
+# - Different options that do not influence the compilation only the Kani flow.
+# - Different options that do not influence the compilation only the CBMC execution.
+
+MANIFEST=lib/Cargo.toml
+OUT_DIR=target
+
+# Expects two arguments: "kani arguments" "output_file"
+function check_kani {
```

Comment:
It might be better to put this function in a different file and source it in each of the scripts to avoid repetition.

Reply:
I totally agree, but this compiletest mode fails if it finds any subdirectory that is not a test. So to reuse this file, I would have to define the function inside one test folder and make all the other tests point to that folder.

So I decided to keep it simple and just use redundancy for now.

Reply:
OK